### PR TITLE
First version of Configs

### PR DESCRIPTION
**Pull request by sskoczyl**
Task: #4

### CODE REVIEW

**Review by janwadolowski — APPROVED** [review]

File: src/sutta_publisher/shared/config.py [hunk 1]
```diff
@@ -5,19 +5,107 @@
...
+        )
+
+        return publication_details
+
+    @classmethod
+    def _get_edition_for_publication(cls, publication: dict) -> dict:
+        """
+        Function to be updated when editions files will be uploaded to bilara-data repo
+        """
+
```
Comment (alple) [inline src/sutta_publisher/shared/config.py:99]: If you assume `scpub3` as the example publication you can make this also to work with examples from https://github.com/suttacentral/bilara-data/tree/sujato-editions/.editions/en/sujato/mn

File: src/sutta_publisher/shared/config.py [hunk 2]
```diff
@@ -5,19 +5,107 @@
...
+    main_toc_depth: str
+    secondary_toc: bool
+    number_of_volumes: int
+    volumes: list
+
+
 @dataclass(frozen=True)
 class Config:
-    bilara_data_url = "https://github.com/suttacentral/bilara-data"
+    PUBLICATION_JSON_RAW_URL = (
```
Comment (alple) [inline src/sutta_publisher/shared/config.py:35]: This one is for internal config use only so
```suggestion
    _PUBLICATION_JSON_RAW_URL = (
```
Also, I'd put at the top of the class the most important/interesting things. The other options are more interesting, so this boring constant I'd put after the others.

File: src/sutta_publisher/shared/config.py [hunk 3]
```diff
@@ -5,19 +5,107 @@
...
+    translation_subtitle: str
+
+    license_type: str
+    license_abbreviation: str
+    license_url: str
+    license_statement: str
+    first_published: str
+
+    editions_url: str
+    editions: EditionsConfig
```
Comment (alple) [inline src/sutta_publisher/shared/config.py:66]: not sure if we need all of it. I'd leave only 
```
creator_name
translation_title
translation_subtitle
root_title
```
also `editions` but that should be `editions: List[EditionsConfig]`

File: tests/unit/test_config.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import pytest
+
+from sutta_publisher.shared.config import Config
+
+
+@pytest.mark.vcr
+def test_config_for_publication(scpub1_data):
+    pub_config = Config.from_publication("scpub1")
+
+    assert asdict(pub_config) == scpub1_data
```
Comment (alple) [inline tests/unit/test_config.py:12]: Nice! and I'd go with explicit checks. `scpub1_data` - it really can bite us when it starts to depend on other tests and json from the repo will change, hence:
```suggestion
    assert pub_config.creator_name == "Bhikkhu Sujato"
    assert pub_config.translation_title == "Middle Discourses"
    ...
```
Those are tests so we want them to be verbose and simple

File: tests/unit/test_config.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from dataclasses import asdict
+
+import pytest
+
+from sutta_publisher.shared.config import Config
+
+
+@pytest.mark.vcr
+def test_config_for_publication(scpub1_data):
```
Comment (alple) [inline tests/unit/test_config.py:9]: Names of tests should follow the convention `test_should_...` so for example 
```suggestion
def test_should_create_config_file():
```

File: src/sutta_publisher/shared/config.py (hunk 3 above)
Comment (sskoczyl) [reply src/sutta_publisher/shared/config.py:66]: Ok, I removed most of this fields. Apart from those mentioned by you, I left `publication_number` so publication could be easily identified. `source_url` and 'editions_url` also can become handy when we would like to download contents of publication or editions.

Comment (alple) [reply src/sutta_publisher/shared/config.py:66]: `editions_url` - it's needed for the config internally yes, but when it's loaded we won't be needing that in application anymore.
with `source_url` also I doubt we will need it - we are going to get all our data (:crossed_fingers: ) from bilara endpoint so also seems like a dead property

File: src/sutta_publisher/shared/config.py (hunk 1 above)
Comment (sskoczyl) [reply src/sutta_publisher/shared/config.py:99]: I think its better to wait when there will be final and structurized version of editions.

**Comment by alple** [comment]
Done in #21

**Pull request closed without merging**